### ISSUE TITLE
Clarify Core vs. Extended Attributes in AARC Attribute Profile

### ISSUE THREAD

**Issue by NicolasLiampotis** [aarc-community/architecture-guidelines#15]
This issue proposes improving the clarity around core/mandatory and optional attributes within [AARC-G056](https://docs.google.com/document/d/1jO1X7GSXWf_R604j5LXinr37ZUf96YVdIXSvjpS_Vyk/edit).

The document currently lacks clear distinction between core/mandatory/essential vs. optional/non-essential attributes.

The document should explicitly define core/mandatory/essential and optional/non-essential attributes and clarify their purpose.

Possible Solutions:
- Consider using tables or separate sections to clearly differentiate between the two types of attributes, or
- Create a new, separate document specifically for optional/non-essential attributes.

**Comment by NicolasLiampotis** [comment]
We distinguish between the following profiles:

- **Core**: This attribute is part of the core profile and is considered essential for basic functionality.
- **Extended**: This attribute is part of the extended profile and may not be supported by all AARC compliant proxies. Its inclusion provides additional information about the subject but is not mandatory.

We propose the following classification for profile attributes:

* Subject Identifier
  * Public Subject Identifier --> Core
  * Pairwise Subject Identifier --> Core
* Name
  * Display Name --> Core
  * Given Name --> Core
  * Family Name --> Core
* Email
  * Email Address --> Core
  * Email verification status  --> _TBD_
* Organisation	
  * Organisation Display Name --> Core
  * Organisation Domain --> Core
  * Organisation’s Country --> Extended
  * Organisation compliance with policies --> Extended
* Affiliation
  * Affiliation within Home Organisation --> Core
  * Affiliation within Community/Research Infrastructure --> _TBD_
* Assurance --> Core
* Group and Role information	--> Core
* Resource Capabilities --> Core
* External Subject Identifiers
  * External subject identifiers --> Extended
  * ORCID --> _TBD_
* SSH Public keys --> Extended
* Certificates
  * Certificate Subject DNs --> Extended
  * Certificate Issuer DNs --> Extended
* Subject Type --> _TBD_
* Subject Status --> Extended
* Agreement to Policies --> Extended
* Authenticating Authorities --> Extended

**Comment by NicolasLiampotis** [comment]
The classification of the attributes has been updated as follows:

<table>
  <tr>
    <th>Attribute Category</th>
    <th>Attribute Name</th>
    <th>Profile (Core or Extended)</th>
  </tr>
  <tr>
    <td rowspan="2"><strong>Subject Identifier</strong></td>
    <td>Public Subject Identifier</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Pairwise Subject Identifier</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="3"><strong>Name</strong></td>
    <td>Display Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Given Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Family Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>Email</strong></td>
    <td>Email Address</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Email verification status</td>
    <td>TBD (see #28)</td>
  </tr>
  <tr>
    <td rowspan="4"><strong>Organisation</strong></td>
    <td>Organisation Display Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Organisation Domain</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Organisation's Country</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>Organisation Compliance with Policies</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>Affiliation</strong></td>
    <td>Affiliation within Home Organisation</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Affiliation within Community/Research Infrastructure</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Assurance</strong></td>
    <td>Assurance</td>
    <td>Core</td>
  </tr>
  <tr>
    <td><strong>Group and Role information</strong></td>
    <td>Group and Role information</td>
    <td>Core</td>
  </tr>
  <tr>
    <td><strong>Resource Capabilities</strong></td>
    <td>Resource Capabilities</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>External Subject Identifiers</strong></td>
    <td>External Subject Identifiers</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>ORCID</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>SSH Public keys</strong></td>
    <td>SSH Public keys</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>Certificates</strong></td>
    <td>Certificate Subject DNs</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>Certificate Issuer DNs</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Subject Type</strong></td>
    <td>Subject Type</td>
    <td>Core</td>
  </tr>
  <tr>
    <td><strong>Subject Status</strong></td>
    <td>Subject Status</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Agreement to Policies</strong></td>
    <td>Agreement to Policies</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Authenticating Authorities</td>
    <td>Authenticating Authorities</td>
    <td>Extended</td>
  </tr>
</table>

Blocked by #28

**Comment by marcvs** [comment]
I wonder:
- how `subject type` made it into "core"
- why `Affiliation withon Community` went to "extended"

**Comment by NicolasLiampotis** [comment]
> * how `subject type` made it into "core"
 
The subject type became a "core" attribute to help relying parties distinguish between physical persons, robots/service accounts, which is important for proper authorisation, access control and policy enforcement. Although there is no standard SAML/OIDC attribute for this, SCIM’s UserType is somewhat related but not a perfect match (UserType is actually closer to what we call Affiliation within the Home Organisation). Another option could be an assurance value under the aarc-community.org domain. Still if no compelling use case for this exists, the subject type could be moved back to the extended profile.

> * why `Affiliation withon Community` went to "extended"

- The affiliation within the community can already be expressed with VO/group entitlements: This means that membership or role within a community can be conveyed using attributes associated with specific Virtual Organisations (VOs) or groups related to the community. So what would be the reason for defining a different attribute for the same information?

- Compatibility with the R&S entity category: The eduPersonScopedAffiliation attribute was initially chosen for its alignment with the REFEDS Research and Scholarship (R&S) entity category. However, the eduPersonScopedAffiliation attribute is not suitable for Infrastructure Proxies: The main issue with eduPersonScopedAffiliation is its requirement for Infrastructure Proxies to handle all the possible scopes of the Community Attribute Authorities (AAs). If an Infrastructure Proxy cannot publish all these scopes, relying parties connected to it might ignore the attribute values due to scope checking. This issue can be avoided by using the voPersonScopedAffiliation attribute which does not require scope checking.

- But still, if there is no real use case for the affiliation within the community that is not already covered by VO/group entitlements, then the attribute should remain part of the extended profile. In other words, if the existing VO/group entitlements are sufficient for expressing community affiliation and no additional functionality is needed, there is little justification for implementing a separate attribute for this purpose.

**Comment by NicolasLiampotis** [comment]
The classification of the attributes has been updated as follows to be based on the Personalized Access Entity Category, plus the community-managed authZ info (groups and capabilities):

<table>
  <tr>
    <th>Attribute Category</th>
    <th>Attribute Name</th>
    <th>Profile (Core or Extended)</th>
  </tr>
  <tr>
    <td rowspan="2"><strong>Subject Identifier</strong></td>
    <td>Public Subject Identifier</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Pairwise Subject Identifier</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="3"><strong>Name</strong></td>
    <td>Display Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Given Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Family Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="1"><strong>Email</strong></td>
    <td>Email Address</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="4"><strong>Organisation</strong></td>
    <td>Organisation Display Name</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Organisation Domain</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Organisation's Country</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>Organisation Compliance with Policies</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>Affiliation</strong></td>
    <td>Affiliation within Home Organisation</td>
    <td>Core</td>
  </tr>
  <tr>
    <td>Affiliation within Community/Research Infrastructure</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Assurance</strong></td>
    <td>Assurance</td>
    <td>Core</td>
  </tr>
  <tr>
    <td><strong>Group and Role information</strong></td>
    <td>Group and Role information</td>
    <td>Core</td>
  </tr>
  <tr>
    <td><strong>Resource Capabilities</strong></td>
    <td>Resource Capabilities</td>
    <td>Core</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>External Subject Identifiers</strong></td>
    <td>External Subject Identifiers</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>ORCID</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>SSH Public keys</strong></td>
    <td>SSH Public keys</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td rowspan="2"><strong>Certificates</strong></td>
    <td>Certificate Subject DNs</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td>Certificate Issuer DNs</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Subject Type</strong></td>
    <td>Subject Type</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Subject Status</strong></td>
    <td>Subject Status</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Agreement to Policies</strong></td>
    <td>Agreement to Policies</td>
    <td>Extended</td>
  </tr>
  <tr>
    <td><strong>Authenticating Authorities</td>
    <td>Authenticating Authorities</td>
    <td>Extended</td>
  </tr>
</table>
